### ISSUE TITLE
Build is failing on libssh_threads.so

### ISSUE THREAD

**Issue by Cellebyte** [CiscoDevNet/ydk-gen#826]
Issue tracker is ONLY used for reporting bugs. Please use the [YDK Community](https://communities.cisco.com/community/developer/ydk) for any support issues.

## Expected Behavior

  pip install ydk

works after installing all required libraries under Arch Linux

Note: libssh 0.8.0 and later does not support separate threading library which is required for YDK. Please use libssh versions older than 0.8.0.

## Current Behavior

Exception is raised on missing libssh_threads.so
Same Issue was resolved here https://github.com/artyom-poptsov/guile-ssh/issues/9

```
Collecting ydk
  Using cached https://files.pythonhosted.org/packages/0f/bf/0dc4dfd7038506f6852175054dc3da7942b717b3e7902c66deebb19c2b66/ydk-0.7.3.tar.gz
Requirement already satisfied: pybind11>=2.1.1 in /home/cellebyte/git/selfnet/prometheus_junos_exporter/env/lib/python3.7/site-packages (from ydk) (2.2.4)
Building wheels for collected packages: ydk
  Running setup.py bdist_wheel for ydk ... error
  Complete output from command /home/cellebyte/git/selfnet/prometheus_junos_exporter/env/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-dn80wctk/ydk/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-7zo0tycp --python-tag cp37:
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/ydk
  copying ydk/__init__.py -> build/lib.linux-x86_64-3.7/ydk
  copying ydk/exthook.py -> build/lib.linux-x86_64-3.7/ydk
  creating build/lib.linux-x86_64-3.7/ydk/logging
  copying ydk/logging/__init__.py -> build/lib.linux-x86_64-3.7/ydk/logging
  creating build/lib.linux-x86_64-3.7/ydk/providers
  copying ydk/providers/__init__.py -> build/lib.linux-x86_64-3.7/ydk/providers
  copying ydk/providers/codec_provider.py -> build/lib.linux-x86_64-3.7/ydk/providers
  creating build/lib.linux-x86_64-3.7/ydk/path
  copying ydk/path/__init__.py -> build/lib.linux-x86_64-3.7/ydk/path
  creating build/lib.linux-x86_64-3.7/ydk/types
  copying ydk/types/py_types.py -> build/lib.linux-x86_64-3.7/ydk/types
  copying ydk/types/__init__.py -> build/lib.linux-x86_64-3.7/ydk/types
  creating build/lib.linux-x86_64-3.7/ydk/filters
  copying ydk/filters/__init__.py -> build/lib.linux-x86_64-3.7/ydk/filters
  creating build/lib.linux-x86_64-3.7/ydk/ext
  copying ydk/ext/__init__.py -> build/lib.linux-x86_64-3.7/ydk/ext
  creating build/lib.linux-x86_64-3.7/ydk/errors
  copying ydk/errors/error_handler.py -> build/lib.linux-x86_64-3.7/ydk/errors
  copying ydk/errors/__init__.py -> build/lib.linux-x86_64-3.7/ydk/errors
  creating build/lib.linux-x86_64-3.7/ydk/entity_utils
  copying ydk/entity_utils/entity_utils.py -> build/lib.linux-x86_64-3.7/ydk/entity_utils
  copying ydk/entity_utils/__init__.py -> build/lib.linux-x86_64-3.7/ydk/entity_utils
  creating build/lib.linux-x86_64-3.7/ydk/models
  copying ydk/models/__init__.py -> build/lib.linux-x86_64-3.7/ydk/models
  creating build/lib.linux-x86_64-3.7/ydk/services
  copying ydk/services/netconf_service.py -> build/lib.linux-x86_64-3.7/ydk/services
  copying ydk/services/__init__.py -> build/lib.linux-x86_64-3.7/ydk/services
  copying ydk/services/codec_service.py -> build/lib.linux-x86_64-3.7/ydk/services
  copying ydk/services/crud_service.py -> build/lib.linux-x86_64-3.7/ydk/services
  copying ydk/services/executor_service.py -> build/lib.linux-x86_64-3.7/ydk/services
  creating build/lib.linux-x86_64-3.7/ydk/path/sessions
  copying ydk/path/sessions/__init__.py -> build/lib.linux-x86_64-3.7/ydk/path/sessions
  copying ydk/path/sessions/restconf_session.py -> build/lib.linux-x86_64-3.7/ydk/path/sessions
  copying ydk/path/sessions/netconf_session.py -> build/lib.linux-x86_64-3.7/ydk/path/sessions
  running build_ext
  -- The C compiler identification is GNU 8.2.1
  -- The CXX compiler identification is GNU 8.2.1
  -- Check for working C compiler: /usr/bin/cc
  -- Check for working C compiler: /usr/bin/cc -- works
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Check for working CXX compiler: /usr/bin/c++
  -- Check for working CXX compiler: /usr/bin/c++ -- works
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  python version: 3.7m
  -- Found PythonLibs: /usr/lib/libpython3.7m.so (found version "3.7.0")
  pybind11 include: /home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7;/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7
  CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
  Please set them or make sure they are set and tested correctly in the CMake files:
  ssh_threads_location
      linked by target "ydk_" in directory /tmp/pip-install-dn80wctk/ydk
  
  -- Configuring incomplete, errors occurred!
  See also "/tmp/pip-install-dn80wctk/ydk/build/temp.linux-x86_64-3.7/CMakeFiles/CMakeOutput.log".
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-dn80wctk/ydk/setup.py", line 149, in <module>
      zip_safe=False,
    File "/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/lib/python3.7/site-packages/setuptools/__init__.py", line 131, in setup
      return distutils.core.setup(**attrs)
    File "/usr/lib64/python3.7/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/usr/lib64/python3.7/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/usr/lib64/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 202, in run
      self.run_command('build')
    File "/usr/lib64/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib64/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/usr/lib64/python3.7/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/usr/lib64/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/usr/lib64/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/tmp/pip-install-dn80wctk/ydk/setup.py", line 72, in run
      self.build_extension(ext)
    File "/tmp/pip-install-dn80wctk/ydk/setup.py", line 104, in build_extension
      subprocess.check_call([cmake_executable, ext.sourcedir] + cmake_args, cwd=self.build_temp)
    File "/usr/lib64/python3.7/subprocess.py", line 328, in check_call
      raise CalledProcessError(retcode, cmd)
  subprocess.CalledProcessError: Command '['cmake', '/tmp/pip-install-dn80wctk/ydk', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/pip-install-dn80wctk/ydk/build/lib.linux-x86_64-3.7', '-DPYBIND11_INCLUDE=/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7;/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7', '-DPYTHON_VERSION=3.7m', '-DCMAKE_BUILD_TYPE=Release', '-DCOVERAGE=False']' returned non-zero exit status 1.
  
  ----------------------------------------
  Failed building wheel for ydk
  Running setup.py clean for ydk
Failed to build ydk
Installing collected packages: ydk
  Running setup.py install for ydk ... error
    Complete output from command /home/cellebyte/git/selfnet/prometheus_junos_exporter/env/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-dn80wctk/ydk/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-v_u2niww/install-record.txt --single-version-externally-managed --compile --install-headers /home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7/ydk:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/ydk
    copying ydk/__init__.py -> build/lib.linux-x86_64-3.7/ydk
    copying ydk/exthook.py -> build/lib.linux-x86_64-3.7/ydk
    creating build/lib.linux-x86_64-3.7/ydk/logging
    copying ydk/logging/__init__.py -> build/lib.linux-x86_64-3.7/ydk/logging
    creating build/lib.linux-x86_64-3.7/ydk/providers
    copying ydk/providers/__init__.py -> build/lib.linux-x86_64-3.7/ydk/providers
    copying ydk/providers/codec_provider.py -> build/lib.linux-x86_64-3.7/ydk/providers
    creating build/lib.linux-x86_64-3.7/ydk/path
    copying ydk/path/__init__.py -> build/lib.linux-x86_64-3.7/ydk/path
    creating build/lib.linux-x86_64-3.7/ydk/types
    copying ydk/types/py_types.py -> build/lib.linux-x86_64-3.7/ydk/types
    copying ydk/types/__init__.py -> build/lib.linux-x86_64-3.7/ydk/types
    creating build/lib.linux-x86_64-3.7/ydk/filters
    copying ydk/filters/__init__.py -> build/lib.linux-x86_64-3.7/ydk/filters
    creating build/lib.linux-x86_64-3.7/ydk/ext
    copying ydk/ext/__init__.py -> build/lib.linux-x86_64-3.7/ydk/ext
    creating build/lib.linux-x86_64-3.7/ydk/errors
    copying ydk/errors/error_handler.py -> build/lib.linux-x86_64-3.7/ydk/errors
    copying ydk/errors/__init__.py -> build/lib.linux-x86_64-3.7/ydk/errors
    creating build/lib.linux-x86_64-3.7/ydk/entity_utils
    copying ydk/entity_utils/entity_utils.py -> build/lib.linux-x86_64-3.7/ydk/entity_utils
    copying ydk/entity_utils/__init__.py -> build/lib.linux-x86_64-3.7/ydk/entity_utils
    creating build/lib.linux-x86_64-3.7/ydk/models
    copying ydk/models/__init__.py -> build/lib.linux-x86_64-3.7/ydk/models
    creating build/lib.linux-x86_64-3.7/ydk/services
    copying ydk/services/netconf_service.py -> build/lib.linux-x86_64-3.7/ydk/services
    copying ydk/services/__init__.py -> build/lib.linux-x86_64-3.7/ydk/services
    copying ydk/services/codec_service.py -> build/lib.linux-x86_64-3.7/ydk/services
    copying ydk/services/crud_service.py -> build/lib.linux-x86_64-3.7/ydk/services
    copying ydk/services/executor_service.py -> build/lib.linux-x86_64-3.7/ydk/services
    creating build/lib.linux-x86_64-3.7/ydk/path/sessions
    copying ydk/path/sessions/__init__.py -> build/lib.linux-x86_64-3.7/ydk/path/sessions
    copying ydk/path/sessions/restconf_session.py -> build/lib.linux-x86_64-3.7/ydk/path/sessions
    copying ydk/path/sessions/netconf_session.py -> build/lib.linux-x86_64-3.7/ydk/path/sessions
    running build_ext
    -- The C compiler identification is GNU 8.2.1
    -- The CXX compiler identification is GNU 8.2.1
    -- Check for working C compiler: /usr/bin/cc
    -- Check for working C compiler: /usr/bin/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /usr/bin/c++
    -- Check for working CXX compiler: /usr/bin/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    python version: 3.7m
    -- Found PythonLibs: /usr/lib/libpython3.7m.so (found version "3.7.0")
    pybind11 include: /home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7;/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7
    CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
    Please set them or make sure they are set and tested correctly in the CMake files:
    ssh_threads_location
        linked by target "ydk_" in directory /tmp/pip-install-dn80wctk/ydk
    
    -- Configuring incomplete, errors occurred!
    See also "/tmp/pip-install-dn80wctk/ydk/build/temp.linux-x86_64-3.7/CMakeFiles/CMakeOutput.log".
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-dn80wctk/ydk/setup.py", line 149, in <module>
        zip_safe=False,
      File "/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/lib/python3.7/site-packages/setuptools/__init__.py", line 131, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib64/python3.7/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/lib64/python3.7/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/usr/lib64/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/lib/python3.7/site-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib64/python3.7/distutils/command/install.py", line 545, in run
        self.run_command('build')
      File "/usr/lib64/python3.7/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib64/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/usr/lib64/python3.7/distutils/command/build.py", line 135, in run
        self.run_command(cmd_name)
      File "/usr/lib64/python3.7/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib64/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-install-dn80wctk/ydk/setup.py", line 72, in run
        self.build_extension(ext)
      File "/tmp/pip-install-dn80wctk/ydk/setup.py", line 104, in build_extension
        subprocess.check_call([cmake_executable, ext.sourcedir] + cmake_args, cwd=self.build_temp)
      File "/usr/lib64/python3.7/subprocess.py", line 328, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['cmake', '/tmp/pip-install-dn80wctk/ydk', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/pip-install-dn80wctk/ydk/build/lib.linux-x86_64-3.7', '-DPYBIND11_INCLUDE=/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7;/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7', '-DPYTHON_VERSION=3.7m', '-DCMAKE_BUILD_TYPE=Release', '-DCOVERAGE=False']' returned non-zero exit status 1.
    
    ----------------------------------------
Command "/home/cellebyte/git/selfnet/prometheus_junos_exporter/env/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-dn80wctk/ydk/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-v_u2niww/install-record.txt --single-version-externally-managed --compile --install-headers /home/cellebyte/git/selfnet/prometheus_junos_exporter/env/include/site/python3.7/ydk" failed with error code 1 in /tmp/pip-install-dn80wctk/ydk/
```

## Steps to Reproduce

```bash
# Generated by debtap
# Maintainer: CISCO
# Contributor: CISCO
pkgname=libydk
pkgver=0.7.3
pkgrel=1
pkgdesc="YDK Core Library. pcre, libssh, libxml, libxslt, libtool, cmake"
arch=('x86_64')
url="https://github.com/CiscoDevNet/ydk-cpp"
license=('Apache')
groups=('')
depends=('pcre' 'libssh' 'libxml2' 'libxslt' 'libtool' 'cmake' 'curl')
options=('!strip' '!emptydirs')
source_x86_64=("https://devhub.cisco.com/artifactory/debian-ydk/${pkgver}/${pkgname}_${pkgver}-${pkgrel}_amd64.deb")
sha512sums_x86_64=('516942987ee4ea30d2c06af8aca7d812212daf673d5462783289dd2a6c6f108d0bda2d0968c4fee69c860e2cc579a50c298d5befaa8605ff892793a8d18741e0')

package(){

	# Extract package data
	tar xzf data.tar.gz -C "${pkgdir}"

	# Fix directories structure differencies
	cd "${pkgdir}"

	if [[ ${CARCH} == x86_64 ]]; then
		mkdir usr/lib 2> /dev/null; mv usr/local/lib/* usr/lib; rm -rf usr/local/lib
	fi

	mv usr/local/* usr; rm -rf usr/local

	cd ..

}
```
----
```bash
makepkg -si

pip install ydk
```

## Your Script

```
cd /lib
ln -s libssh.so libssh_threads.so
```

## Logs
Enable [logging](http://ydk.cisco.com/py/docs/guides/introduction.html#logging) and post the logs below

```

```

## System Information
* python 3.7
* Arch Linux
* x86_64

**Comment by ygorelik** [comment]
Known issue of libssh-0.8.x. Recommended install libssh-0.7.6 from source or Install libssh-0.8.1-3.

**Comment by Cellebyte** [comment]
Not an option. I linked libssh-threads against libssh and it works.

**Comment by ygorelik** [comment]
Could you please elaborate on how did you "linked libssh-threads against libssh".

**Comment by Cellebyte** [comment]
Only symlinked  libssh-threads on libssh and build the project.

**Comment by valeriodigregorio** [comment]
Build for 0.8.4 failed:
```
checking for libssh/libssh.h... yes
checking for ssh_threads_get_pthread in -lssh_threads... no
configure: error: Missing the libssh_threads library.
make[2]: *** [CMakeFiles/project_libnetconf.dir/build.make:109: project_libnetconf/src/project_libnetconf-stamp/project_libnetconf-configure] Error 1
make[1]: *** [CMakeFiles/Makefile2:169: CMakeFiles/project_libnetconf.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```
even if I have:
```
lrwxrwxrwx 1 root root     11 Aug  5  2019 /lib/libssh.so -> libssh.so.4
lrwxrwxrwx 1 root root     15 Aug  5  2019 /lib/libssh.so.4 -> libssh.so.4.8.1
-rwxr-xr-x 1 root root 834124 Aug  5  2019 /lib/libssh.so.4.8.1
lrwxrwxrwx 1 root root      9 Jun  4 14:31 /lib/libssh_threads.so -> libssh.so
lrwxrwxrwx 1 root root     15 Aug  5  2019 /lib/libssh_threads.so.4 -> libssh.so.4.8.1
lrwxrwxrwx 1 root root     15 Aug  5  2019 /lib/libssh_threads.so.4.8.1 -> libssh.so.4.8.1
```

**Comment by ygorelik** [comment]
On 'fedora' based systems you need to configure LD_LIBRARY_PATH or set CMAKE_LIBRARY_PATH (similar to LD_LIBRARY_PATH, but localized for make), before compiling the YDK code. This is "Semicolon-separated list of directories specifying a search path for the find_library() command". You might also need configure CMAKE_C_INCLUDE_PATH and CMAKE_CPLUS_INCLUDE_PATH for the cmake to find include files.

**Comment by valeriodigregorio** [comment]
Apparently I solved this way:
```bash
sudo ln -s libssh.so /lib/libssh_threads.so
sudo ln -s libssh.so /lib64/libssh_threads.so
```
I was missing second line. Thanks for your reply.

**Comment by ygorelik** [comment]
This method works fine in "fedora" based platform, but for some reason fails in Ubuntu:focal (20.04 LTS). The compilation works fine, but Netconf client gets stalled (apparently gets to endless loop based on CPU load), when trying connect to the server.

**Comment by ygorelik** [comment]
The issue was resolved in YDK release 0.8.6.6 and 0.9.1.2.
